### PR TITLE
LDAP Wizard: increasing server numbers and avoid configID collisions, fixes #15766

### LINKED ISSUE #15766: Adding LDAP server reuses last number in the list
### Steps to reproduce

1. Install 8.1 enterprise
2. Enable LDAP app
3. Navigate to admin page
4. Add a bunch of new LDAP servers (I added 5)
5. Delete some of the middle ones (deleting one will do)
6. Add a new LDAP server

### Expected behaviour

New LDAP server should get the next numerical value available

### Actual behaviour

The next LDAP server created reuses the last number:

p321

### Server configuration

Ubuntu 14.04
PHP 5.5.9
ownCloud Enterprise Edition 8.1 alpha 1 (daily) Build:2015-04-16T03:08:13+00:00 f50798a674bf223e1c8015b9a27287b2ac6c9f1a

### PR DESCRIPTION
**Pull request by blizzz**
How to reproduce:
1. Have a number (y) of LDAP configurations 
2. Delete a few of them
3. Add a new one: new "x. Server" number in the configuration chooser should be (y+1). (Before it was number of available entries + 1).

The other thing:
1. Add a couple of empty(!)  LDAP configurations
2. Change a value in one of them
3. Switch to another one. Before: it seems the empty config was not loaded, but in fact the configID was reused. Now: empty configs are saved upon requests, avoiding such collisions. i.e. the empty config is loaded properly.

Please test and review @jnfrmarks @MorrisJobke @PVince81 @Xenopathic or others, ty

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
asking for unit tests here is pointless - @blizzz time to move to Appframework :speak_no_evil:

**Comment by blizzz** [comment]
>  time to move to Appframework 

it's on my mind, but connected with a overhaul of the whole backend part of the app.

File: apps/user_ldap/js/wizard/wizardTabElementary.js [hunk 1]
```diff
@@ -204,9 +207,17 @@ OCA = OCA || {};
 		 */
 		onNewConfiguration: function(view, result) {
 			if(result.isSuccess === true) {
-				$(view.configChooserID + ' option:selected').removeAttr('selected');
-				var html = '<option value="'+result.configPrefix+'" selected="selected">'+t('user_ldap','{nthServer}. Server', {nthServer: $(view.configChooserID + ' option').length + 1})+'</option>';
-				$(view.configChooserID + ' option:last').after(html);
+				console.log('new config');
```
Comment (DeepDiver1975) [inline apps/user_ldap/js/wizard/wizardTabElementary.js]: :hust:

**Comment by DeepDiver1975** [comment]
LDAP testing please @jnfrmarks @SergioBertolinSG @MorrisJobke THX

**Comment by jnfrmarks** [comment]
@jvillafanez 

Please add this to the test plan.

**Comment by DeepDiver1975** [comment]
> Please add this to the test plan.

just to avoid miss understandings: this change is not yet merged - would be great if this can be tested asap so that we can merge this - THX

**Comment by jnfrmarks** [comment]
Clarifying: @jvillafanez Can you please also help out with validating this PR?  Thanks!

**Comment by jvillafanez** [comment]
Reproduced both issues and fixed in the branch. :+1:

**Comment by DeepDiver1975** [comment]
thx @jvillafanez

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**